### PR TITLE
Show partial renders using points while calculating triangles

### PR DESCRIPTION
**Pull request by uclaros**
## Description
When _Render as a surface (triangulate)_ is enabled for point cloud 2d renderers, show a temporary preview render using points while computing the actual triangles.

### CODE REVIEW

File: src/core/pointcloud/qgspointcloudlayerrenderer.cpp [hunk 1]
```diff
@@ -715,6 +735,14 @@ void QgsPointCloudLayerRenderer::renderTriangulatedSurface( QgsPointCloudRenderC
   painter->drawImage( 0, 0, img );
 }
 
+Qgis::MapLayerRendererFlags QgsPointCloudLayerRenderer::flags() const
+{
+  // when rendering as triangles we still want to show temporary incremental renders as points until
+  // the final triangulated surface is ready, which may be slow
+  // So we request here a preview render image for the temporary incremental updates:
+  return Qgis::MapLayerRendererFlag::RenderPartialOutputs | Qgis::MapLayerRendererFlag::RenderPartialOutputOverPreviousCachedImage;
```
Comment (nyalldawson) [inline src/core/pointcloud/qgspointcloudlayerrenderer.cpp:743]: Shouldn't this be dependent on whether render as triangles is active?

**Review by nyalldawson — COMMENTED** [review]
Thanks! Fwiw, the same approach could be used to draw points incrementally if either of the point z sorting options are active.

File: src/core/pointcloud/qgspointcloudlayerrenderer.cpp (hunk 1 above)
Comment (uclaros) [reply src/core/pointcloud/qgspointcloudlayerrenderer.cpp:743]: True, since we can guarantee that `mRenderer` is only set in the constructor

**Comment by uclaros** [comment]
> Thanks! Fwiw, the same approach could be used to draw points incrementally if either of the point z sorting options are active.

This would require some restructuring of the logic to avoid re-decompression of the blocks for unsorted and sorted rendering. OTOH, with the caching implementation the cost would be quite minimal.

**Review by wonder-sk — APPROVED** [review]

**Pull request merged**